src/fondant/pipeline/lightweight_component.py

```diff
@@ -0,0 +1,38 @@
+import typing as t
+from dataclasses import dataclass
+from functools import wraps
+
+
+@dataclass
+class Image:
+    base_image: t.Optional[str] = "fondant:latest"
+    extra_requires: t.Optional[t.List[str]] = None
+    script: t.Optional[str] = None
```

Comment:
For every runner, since the script is not baked into the image. In the docker & KfP runner, we'll probably include the script in the entrypoint instead of uploading it separately though.